fix!: New poseidon2 chunks implementation

sirasistant opened the pull request:
Rewrite the poseidon2 chunks implementation, the previous one was hard to read due to intertwined trailing zeroes checks and had some issues with that. The new one is less gates and should be more readable after separating trailing zero checks and business logic

sirasistant commented:
cc @MirandaWood to double check the new impl :pray:

sirasistant commented:
Gates diff
<img width="2317" height="780" alt="image" src="https://github.com/user-attachments/assets/01496d68-f551-497c-a76c-3160b3e9b7fb" />

MirandaWood reviewed (commented):
Great!! So much cleaner and glad gates have gone down. Only nit could be to rename `poseidon2_absorb_chunks_existing_sponge` -> `poseidon2_absorb_in_chunks_existing_sponge`, but not really necessary 🚀

LeilaWang reviewed (approved):
Nice fix and refactoring 🙌

sirasistant committed:
fix!: New poseidon2 chunks implementation

Rewrite the poseidon2 chunks implementation, the previous one was hard to read due to intertwined trailing zeroes checks and had some issues with that. The new one is less gates and should be more readable after separating trailing zero checks and business logic
